photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/GaussianProcessSearch.scala

```diff
@@ -92,17 +92,33 @@ class GaussianProcessSearch[T](
         // hyperparameter spaces.
         val kernel = new Matern52
 
+        //Finding the overall bestEval
+        val currentMean =  mean(evals)
+        val overallBestEval = if(evaluator.betterThan(priorBestEval, bestEval - currentMean)) {
+          priorBestEval
+        } else {
+          bestEval - currentMean
+        }
+
         // Expected improvement transformation
-        val transformation = new ExpectedImprovement(evaluator, bestEval)
+        val transformation = new ExpectedImprovement(evaluator, overallBestEval)
 
         val estimator = new GaussianProcessEstimator(
           kernel = kernel,
-          normalizeLabels = true,
+          normalizeLabels = false,
           noisyTarget = noisyTarget,
           predictionTransformation = Some(transformation),
           seed = seed)
 
-        val model = estimator.fit(points, evals)
+        //Union of points and evals with priorData
+        val (overallPoints, overallEvals) = if(priorObservedPoints.isDefined) {
+          (DenseMatrix.vertcat(points, priorObservedPoints.get),
+            DenseVector.vertcat(evals - currentMean, priorObservedEvals.get))
+        } else {
+          (points, evals - currentMean)
+        }
```

Comment:
It would be more idiomatic to set this up as a pattern match, e.g.:

```scala
val (overallPoints, overallEvals) = (priorObservedPoints, priorObservedEvals) match {
  case (Some(priorPoints), Some(priorEvals)) => (
    DenseMatrix.vertcat(points, priorPoints),
    DenseVector.vertcat(evals - currentMean, priorEvals))
  case _ =>
    (points, evals - currentMean)
}
```

In general, when you find yourself using the `get` function to unwrap an `Option`, there's usually a better way to handle it.

Reply:
Thanks Josh. I've made the changes accordingly.